cloudinit/handlers/jinja_template.py

```diff
@@ -4,15 +4,19 @@
 import os
 import re
 from errno import EACCES
-from typing import Optional
+from typing import Optional, Type
```

Comment:
I agree this version is more compact and legible, but we need an extra line to avoid flake8 think we are doing logic before the subsequent imports and mark them as `E402 module level import not at top of file`. That's why we need to add the `TYPE_CHECKING` check:

```python
if TYPE_CHECKING:
    JUndefinedError: Type[Exception]
try:
    from jinja2.exceptions import UndefinedError as JUndefinedError
    from jinja2.lexer import operator_re
except ImportError:
    # No jinja2 dependency
    JUndefinedError = Exception
    operator_re = re.compile(r"[-.]")
```

I am okay with both syntax.

Reply:
Yours is ok, or we could move the try to the bottom of the imports. isort and black shouldn't complain because they're in a try/except. Either way is fine. You can choose.

Reply:
Your version moved down. Thanks!